app/services/compute_resource_host_associator.rb

```diff
@@ -15,6 +15,12 @@ def associate_hosts
     end
   end
 
+  def associate_hosts_to_vm(vm)
```

Comment:
Looks a bit redundant.
You can reuse the original `associate_hosts` method:

``` ruby
  def associate_hosts(vms = compute_resource.vms(:eager_loading => true))
    vms.each do |vm|
      if Host.for_vm(compute_resource, vm).empty?
        associate_vm(vm)
      end
    end
  end
```

And the usage for a specific vm would be:
``` ruby
@vm = @compute_resource.find_vm_by_uuid(params[:vm_id])
associator.associate_hosts([@vm])
```

I think this way you can get rid of the new route altogether if you add :vm_id as an optional parameter to `associate`.